File: numeric/primes500/primes.r

get_primes <- function() {
  primes <- list(2)
  n <- 3

  for (j in 1:500) {
    primes <- append(primes, n)

    is_prime <- FALSE
    while (!is_prime) {
      n <- n + 2

      q <- 9999
      r <- 1
      t <- 0
      k <- 2
      while ((r != 0) && (q > t)) {
        t <- primes[[k]]
        q <- n / t
        r <- n %% t
        k <- k + 1
      }
      is_prime <- (r != 0) && (q <= t)
    }
  }
  primes
}

print_primes <- function(primes) {
  print("First Five Hundred Primes")
  for (i in 1:50) {
    print(sprintf("     %04d %04d %04d %04d %04d %04d %04d %04d %04d %04d",
                  primes[[i]], primes[[50 + i]], primes[[100 + i]],
                  primes[[150 + i]], primes[[200 + i]], primes[[250 + i]],
                  primes[[300 + i]], primes[[350 + i]], primes[[400 + i]],
                  primes[[450 + i]]))
  }
}

print_primes(get_primes())

File: numeric/primes500/primes.php
<?php
function prime_sieve() {
    yield 2;
    $cache = [2];
    $candidate = 3;
    while (true) {
        if (array_all($cache, function ($prime) use ($candidate) {
            return $candidate % $prime != 0; })) {
            yield $candidate;
            $cache[] = $candidate;
        }

        $candidate += 2;
    }
}

function get_primes($count) {
    $primes = [];
    $n = 0;
    foreach (prime_sieve() as $prime) {
        $primes[] = $prime;
        $n++;
        if ($n >= $count) {
            break;
        }
    }
    return $primes;
}

function print_primes($primes) {
    echo "First Five Hundred Primes\n";
    for ($i = 0; $i < 50; ++$i) {
        echo sprintf("     %04d %04d %04d %04d %04d %04d %04d %04d %04d %04d\n",
            $primes[$i], $primes[$i + 50], $primes[$i + 100], $primes[$i + 150],
            $primes[$i + 200], $primes[$i + 250], $primes[$i + 300],
            $primes[$i + 350], $primes[$i + 400], $primes[$i + 450]);
    }
}

print_primes(get_primes(500));
?>